.gitignore

```diff
@@ -23,3 +23,4 @@
 /.pydevproject
 /.idea
 /*.iml
+*.swp
```

Comment:
The reason is that when you switch branches in git, it leaves behind ignored files. So, we've stopped doing global ignores, and instead have been doing "rooted" ignores (I don't know if that's the official term... it's just what I'm calling ignore expressions beginning with "/"). That way, when you switch branches, you can see what was left behind (most frequently, "target/" directories from previous builds for modules which exist in another branch, but not the current one). This allows you to do more targeted `git clean` executions with `git clean -fd` instead of needing to do `git clean -fdx`.

If we do global ignores (with the prefixing "/"), some maven plugins (assembly, rat, etc.) may treat those extra files as source files, because they don't exist in a current module's "target/" directory (other source directories could, after all, also use the name "target").

Keeping the `.gitignore` files rooted to each module helps us avoid ignoring files which could be problematic if ignored globally.